am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/importer/DefaultPermissionIntegrationTest.java

```diff
@@ -25,89 +27,93 @@
     }
 
     @Test
     void shouldNotBeAbleToCreateDefaultPermissionWhenRoleDoesNotExist() {
         assertThatExceptionOfType(PersistenceException.class)
-            .isThrownBy(() -> service.grantDefaultPermission(createDefaultPermissionGrant(ImmutableSet.of(READ))))
+            .isThrownBy(() -> service.grantDefaultPermission(createDefaultPermissionGrant(resourceDefinition, ImmutableSet.of(READ))))
             .withMessageContaining("(role_name)=(Solicitor) is not present in table \"roles\"");
     }
 
     @Test
     void shouldAddNewEntryIntoDatabaseWhenUniqueEntry() {
         service.addRole(ROLE_NAME, RESOURCE, PUBLIC, ROLE_BASED);
-        service.grantDefaultPermission(createDefaultPermissionGrant(ImmutableSet.of(READ)));
+        service.grantDefaultPermission(createDefaultPermissionGrant(resourceDefinition, ImmutableSet.of(READ)));
 
-        assertThat(databaseHelper.countDefaultPermissions(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME,
+        assertThat(databaseHelper.countDefaultPermissions(resourceDefinition,
             ROOT_ATTRIBUTE.toString(), ROLE_NAME, Permissions.sumOf(ImmutableSet.of(READ)))).isEqualTo(1);
 
-        assertThat(databaseHelper.getResourceAttribute(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME,
+        assertThat(databaseHelper.getResourceAttribute(resourceDefinition,
             ROOT_ATTRIBUTE.toString(), PUBLIC)).isNotNull();
     }
 
     @Test
     void shouldOverwriteExistingRecordWhenEntryIsAddedASecondTime() {
         service.addRole(ROLE_NAME, RESOURCE, PUBLIC, ROLE_BASED);
-        service.grantDefaultPermission(createDefaultPermissionGrant(ImmutableSet.of(READ)));
-        service.grantDefaultPermission(createDefaultPermissionGrant(ImmutableSet.of(CREATE)));
+        service.grantDefaultPermission(createDefaultPermissionGrant(resourceDefinition, ImmutableSet.of(READ)));
+        service.grantDefaultPermission(createDefaultPermissionGrant(resourceDefinition, ImmutableSet.of(CREATE)));
 
-        assertThat(databaseHelper.countDefaultPermissions(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME,
+        assertThat(databaseHelper.countDefaultPermissions(resourceDefinition,
             ROOT_ATTRIBUTE.toString(), ROLE_NAME, Permissions.sumOf(ImmutableSet.of(CREATE)))).isEqualTo(1);
 
-        assertThat(databaseHelper.getResourceAttribute(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME,
+        assertThat(databaseHelper.getResourceAttribute(resourceDefinition,
             ROOT_ATTRIBUTE.toString(), PUBLIC)).isNotNull();
     }
 
     @Test
     void shouldRemoveAllEntriesFromTablesWhenValuesExist() {
         service.addRole(ROLE_NAME, RESOURCE, PUBLIC, ROLE_BASED);
-        service.addResourceDefinition(createResourceDefinition(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME));
-        service.addResourceDefinition(createResourceDefinition(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME + 2));
+        service.addResourceDefinition(createResourceDefinition(serviceName, RESOURCE_TYPE, RESOURCE_NAME));
```

Comment:
Why `resourceDefinition` field is not used here?

Reply:
👍